File: sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/IMarketplacePresenter.java
package com.kin.ecosystem.marketplace.presenter;

import com.kin.ecosystem.base.IBasePresenter;
import com.kin.ecosystem.core.network.model.Offer.OfferType;
import com.kin.ecosystem.main.INavigator;
import com.kin.ecosystem.marketplace.view.IMarketplaceView;

public interface IMarketplacePresenter extends IBasePresenter<IMarketplaceView> {

    void onStart();

    void onStop();

    void getOffers();

    void onItemClicked(int position);

    void showOfferActivityFailed();

    void backButtonPressed();

    void setNavigator(INavigator navigator);

	void closeClicked();

    void myKinCLicked();
}

File: sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/IMarketplacePresenter.kt
package com.kin.ecosystem.marketplace.presenter

import com.kin.ecosystem.base.IBaseFragmentPresenter
import com.kin.ecosystem.base.IBasePresenter
import com.kin.ecosystem.main.INavigator
import com.kin.ecosystem.marketplace.view.IMarketplaceView

interface IMarketplacePresenter : IBaseFragmentPresenter<IMarketplaceView> {

    fun onResume()

    fun onPause()

    fun getOffers()

    fun onItemClicked(position: Int)

    fun showOfferActivityFailed()

    fun closeClicked()

    fun myKinCLicked()
}
